server/server_test.go

```diff
@@ -1875,55 +1875,67 @@ func TestDataMetrics(t *testing.T) {
 
 	f := newFixture(t)
 
-	req := newReqV1(http.MethodPost, "/data?metrics", "")
-	f.reset()
-	f.server.Handler.ServeHTTP(f.recorder, req)
-
-	var result types.DataResponseV1
-
-	if err := util.NewJSONDecoder(f.recorder.Body).Decode(&result); err != nil {
-		t.Fatalf("Unexpected JSON decode error: %v", err)
-	}
-
-	// Test some basic well-known metrics.
-	expected := []string{
+	testDataMetrics(t, f, "/data?metrics", []string{
+		"counter_server_query_cache_hit",
+		"timer_rego_input_parse_ns",
+		"timer_rego_load_bundles_ns",
+		"timer_rego_load_files_ns",
+		"timer_rego_module_parse_ns",
 		"timer_rego_query_parse_ns",
 		"timer_rego_query_compile_ns",
 		"timer_rego_query_eval_ns",
 		"timer_server_handler_ns",
-	}
+	})
 
-	for _, key := range expected {
-		if result.Metrics[key] == nil {
-			t.Fatalf("Expected non-zero metric for %v but got: %v", key, result)
-		}
-	}
+	testDataMetrics(t, f, "/data?metrics", []string{
+		"counter_server_query_cache_hit",
+		"timer_rego_input_parse_ns",
+		"timer_rego_query_parse_ns",
```

Comment:
I was surprised that `timer_rego_query_parse_ns` shows up. It's just odd because the query compile and module parse/compile timers are not here. Since there is so little time spent on query parsing, I'm guessing is just a time start/deferred stop case. Is it tough to remove this?

Reply:
Yea, same thing happens for the `timer_rego_input_parse_ns`. I'll see if we can refactor things over in rego.go to remove it.

Reply:
Fixed a bunch of them in the latest updates.